ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -337,8 +337,8 @@ public CompletableFuture<PutResult> put(NamedBlobRecord record, NamedBlobState s
           PutResult putResult = null, putResultV2 = null;
           try {
             putResult = run_put(record, accountId, containerId, connection);
-          } catch (Exception eOld) {
-            if (!config.dbRelyOnNewTable) {
+          } catch (RestServiceException eOld) {
+            if (!config.dbRelyOnNewTable || eOld.getErrorCode() == RestServiceErrorCode.Conflict) {
```

Comment:
whenever the boolean grows long, make a helper. Its likely it'll grow.
i don't know what this expression means but a readable helper would instantly convey the meaning.
I can think of the following:
```
void onConflictError(eOld) throws RestServiceException {
  // your logic to throw here
}
```

Reply:
And please make a new catch block for this exception.

Reply:
```
catch (NewException) {
 onNewException()
}
```
something like this

Reply:
This is outdated, I updated per Justin's comment in 
2e39a00. Can you take another look?